src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs

```diff
@@ -31,18 +40,32 @@ private MediaTypeHeaderValue()
             // Used by the parser to create a new instance of this type.
         }
 
+        /// <summary>
+        /// Initializes a <see cref="MediaTypeHeaderValue"/> instance.
+        /// </summary>
+        /// <param name="mediaType">The <see cref="StringSegment"/> with the media type.
+        /// The media type constructed here must not have any trailing parameters.</param>
         public MediaTypeHeaderValue(StringSegment mediaType)
         {
-            CheckMediaTypeFormat(mediaType, "mediaType");
+            CheckMediaTypeFormat(mediaType, MediaTypeString);
             _mediaType = mediaType;
         }
 
+        /// <summary>
+        /// Initializes a <see cref="MediaTypeHeaderValue"/> instance.
+        /// </summary>
+        /// <param name="mediaType">The <see cref="StringSegment"/> with the media type.
+        /// The media type constructed here must not have any trailing parameters.</param>
+        /// <param name="quality">The <see cref="double"/> with the quality of the media type.</param>
         public MediaTypeHeaderValue(StringSegment mediaType, double quality)
             : this(mediaType)
         {
             Quality = quality;
         }
 
+        /// <summary>
+        /// Gets the charset parameter of the <see cref="MediaTypeHeaderValue"/> if it has one.
```

Comment:
I realize these are copy-paste but these doc strings are meh. Usually you'd say something like:
```
Gets or sets the value of the charset parameter. Will return <see cref="StringSegment.Empty"> if the charset parameter is not present.
```

I'm not going to comment on all of these, but take care that they say 'gets' or 'gets or sets' as appropriate.